src/api.js

```diff
@@ -1362,9 +1364,12 @@ api.declare({
     res.once('close', accept);
   });
 
-  let result = await this.workClaimer.claim(
-    provisionerId, workerType, workerGroup, workerId, count, aborted,
-  );
+  let [result] = await Promise.all([
+    this.workClaimer.claim(
+      provisionerId, workerType, workerGroup, workerId, count, aborted,
+    ),
+    this.workerInfo.provisionerSeen(provisionerId),
```

Comment:
you would call `monitor.reportError`

Not having the promise `await`ed or `.catch()`'d seems like a bad idea, especially as it completely drops on the floor if something goes wrong.  What about storing a reference to the promise and waiting for it after the res.reply?